dbt_copilot_helper/commands/dns.py

```diff
@@ -749,5 +756,301 @@ def assign(app, domain_profile, project_profile, svc, env):
+                        bold=True,
+                    ),
+                )
+
+            # Get the current rule so we can update host header with new domain.
+            for rule in response["Rules"]:
+                save = True
+                rule_arn = rule["RuleArn"]
+                conditions = rule["Conditions"]
+
+                if conditions:
+                    for cond in conditions:
+                        # Only check if the condition is using the host header
+                        if cond["Field"] == "host-header":
+                            values_string = ";".join(cond["Values"])
+                            click.echo(
+                                click.style("Domains currently configured: ", fg="yellow")
+                                + click.style(f"{values_string}", fg="white", bold=True),
+                            )
+                            cdn_domain = input(
+                                f"Enter in domain you wish to add (or press Enter to skip): "
+                            )
+                            if not cdn_domain:
+                                save = False
+                                break
+
+                            # When adding exit if domain already exists.
+                            if cdn_domain in cond["Values"]:
+                                click.echo(
+                                    click.style(f"{cdn_domain} already exists, exiting", fg="red"),
+                                )
+                                save = False
+                                break
+
+                            cond["HostHeaderConfig"]["Values"].append(cdn_domain)
+
+                            # Remove unwanted config values as not needed in update.
+                            cond.pop("Values", None)
+
+                        if cond["Field"] == "path-pattern":
+                            cond.pop("PathPatternConfig", None)
+
+                    if save:
+                        update_rule(
+                            False,
+                            listener,
+                            conditions,
+                            rule_arn,
+                            elb_client,
+                            acm_client,
+                            cdn_domain,
+                        )
+
+            break
+
+
+@cdn.command(name="delete")
+@click.option(
+    "--project-profile", help="AWS account profile name for certificates account", required=True
+)
+@click.option("--env", help="AWS Copilot environment name", required=True)
+@click.option("--app", help="Application Name", required=True)
+@click.option("--svc", help="Service Name", required=True)
+# @click.option("--delete", help="Delete the CDN domain", is_flag=True)
+# Feature to be added at later date if needed, there shouldn't be a need to remove all domains
+# @click.option("--force", help="Force remove", is_flag=True)
+def cdn_delete(project_profile, env, app, svc, force=False):
```

Comment:
A significant amount of this method appears to be the same as in the `assign`, and some perhaps in the `list` method.

I think we can probably separate that out to method(s) to reduce duplication. I'd be happy to pair with you on that.